category/basic.lean

```diff
@@ -112,6 +112,29 @@ def mtry {α} (x : F α) : F unit := (x $> ()) <|> pure ()
 
 end alternative
 
+namespace sum
+
+variables {e : Type v}
+
+protected def bind {α β} : e ⊕ α → (α → e ⊕ β) → e ⊕ β
+| (inl x) _ := inl x
+| (inr x) f := f x
+
+instance : monad (sum.{v u} e) :=
+{ pure := @sum.inr e,
+  bind := @sum.bind e }
+
+instance : is_lawful_functor (sum.{v u} e) :=
+by refine { .. }; intros; cases x; refl
```

Comment:
here and in the following proofs: `x` is bound by `intros`, but not named explicitly. If something changes this bound names (because `intros` behaves different, or the class definition changes etc) then the name might change and the proof breaks.

Can you pass explicit names to `intros`?!

Reply:
do you prefer `intros x ; cases x` or `intros ; casesm sum _ _`?